processor/processor.go

```diff
@@ -1350,18 +1367,57 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob, parsedE
 					source,
 				)
 
+				payloadFunc := func() json.RawMessage {
```

Comment:
we are effectively calling this twice. We can memoize it by assigning the result to a variable and returning this variable during subsequent calls